apps/prairielearn/src/pages/studentAssessment/studentAssessment.js

```diff
@@ -21,125 +22,97 @@ import {
 import { getClientFingerprintId } from '../../middlewares/clientFingerprint.js';
 import { checkPasswordOrRedirect } from '../../middlewares/studentAssessmentAccess.js';
 
+import { StudentAssessment } from './studentAssessment.html.js';
+
 const router = express.Router();
 const sql = loadSqlEquiv(import.meta.url);
 
 router.get(
```

Comment:
This router code was extensively re-organized. The original code was confusing and had lots of duplicated logic. I changed the order to allow for early termination in cases where the page itself is not being rendered because of a redirection.